src/script/media/MediaDevicesHandler.ts

```diff
@@ -343,6 +359,34 @@ export class MediaDevicesHandler {
     return screenSources;
   }
 
+  /**
+   * Enumerates all known Media Devices Infos of a System.
+   * @param forceCamera If `forceCamera=true`, then an audio and video track is created to enforce the device access
+   * permissionm
+   * @returns All enumerated MediaDeviceInfos
+   */
+  private async enumerateMediaDevices(forceCamera: boolean = false): Promise<MediaDeviceInfo[]> {
+    // The browser will set the device labels obtained from navigator.mediaDevices.enumerateDevices() to blank strings
+    // if there is no longer an active MediaStream, even if the application was previously temporarily authorized to
+    // access the devices by calling navigator.mediaDevices.getUserMedia().
+
+    // @TODO: Fix Issue
+    // You can use this method to generate a camera stream even in the audio case. This way, the device labels can be displayed in any case.
+    // const constraints = (forceCamera)? {video: true} : {audio: true};
+    // const mediaPromise = navigator.mediaDevices.getUserMedia(constraints)
+    // return mediaPromise.then(stream => {
+    //   if (!stream) {
+    //     return [];
+    //   }
+    //   return window.navigator.mediaDevices.enumerateDevices().then( devices  => {
+    //     stream.getTracks().forEach((t) => t.stop());
+    //     return devices;
+    //   });
+    // });
+
+    return window.navigator.mediaDevices.enumerateDevices();
+  }
```

Comment:
his is how I believe it should be done. However, this would also start a camera for an audio call. I’ve kept the code in place so we can find a better solution.

Reply:
If a camera is started when launching the app or elsewhere, the user will be prompted for permissions. This can be inconvenient if the user does not intend to make a call.

Reply:
That's why the device list is now created only when a call is about to be made or when the device selection is opened.

Reply:
To avoid any conflicts with the old behavior, I never start my own stream. I always use only the currently active streams.